src/praisonai-agents/praisonaiagents/scheduler/runner.py

```
@@ -114,6 +114,74 @@ def mark_run(
+            return None
+        job_id = ref
+        get = getattr(self._store, "get", None)
+        if not (callable(get) and get(ref) is not None):
+            get_by_name = getattr(self._store, "get_by_name", None)
+            if callable(get_by_name):
+                match = get_by_name(ref)
+                if match is not None:
+                    job_id = match.id
+        try:
+            records = get_history(job_id=job_id)
+        except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)